pkg/apis/pipeline/v1alpha2/workspace_validation.go

```diff
@@ -31,19 +39,41 @@ func (b *WorkspaceBinding) Validate(ctx context.Context) *apis.FieldError {
 		return apis.ErrMissingField(apis.CurrentField)
 	}
 
-	// Users should only provide one supported VolumeSource.
-	if b.PersistentVolumeClaim != nil && b.EmptyDir != nil {
-		return apis.ErrMultipleOneOf("workspace.persistentvolumeclaim", "workspace.emptydir")
+	numSources := b.numSources()
+
+	if numSources > 1 {
+		return apis.ErrMultipleOneOf(allVolumeSourceFields...)
 	}
 
-	// Users must provide at least one supported VolumeSource.
-	if b.PersistentVolumeClaim == nil && b.EmptyDir == nil {
-		return apis.ErrMissingOneOf("workspace.persistentvolumeclaim", "workspace.emptydir")
+	if numSources == 0 {
+		return apis.ErrMissingOneOf(allVolumeSourceFields...)
 	}
 
 	// For a PersistentVolumeClaim to work, you must at least provide the name of the PVC to use.
 	if b.PersistentVolumeClaim != nil && b.PersistentVolumeClaim.ClaimName == "" {
 		return apis.ErrMissingField("workspace.persistentvolumeclaim.claimname")
 	}
+
+	// For a ConfigMap to work, you must provide the name of the ConfigMap to use.
+	if b.ConfigMap != nil && b.ConfigMap.LocalObjectReference.Name == "" {
+		return apis.ErrMissingField("workspace.configmap.name")
+	}
```

Comment:
should we have a unit test that tries to set a configmap + something else?

that's quickly starting to balloon out of control unit test permutation wise - i wonder if we can isolate the logic for determining this somehow such that the tests wont need to be updated with every permutation :thinking: